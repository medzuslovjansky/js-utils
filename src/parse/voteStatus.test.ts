import parseVoteStatus from './voteStatus';

describe('parseVoteStatus', () => {
  it.each([
    ['Common', 1],
    ['Regional', 2],
    ['Individual', 3],
    ['Archaic', 4],
    ['Artificial', 5],
    ['Questionable', 9],
    ['Deprecated', 99],
  ])('should return %j for %j', (expected, input) => {
    expect(parseVoteStatus(input)).toBe(expected);
    expect(parseVoteStatus(`${input}`)).toBe(expected);
  });

  it.each([[null], [100], ['A']])(
    'should throw given an unknown value: %j',
    (input: any) => {
      expect(() => parseVoteStatus(input)).toThrowErrorMatchingSnapshot();
      expect(() => parseVoteStatus(`${input}`)).toThrowError();
    },
  );
});
