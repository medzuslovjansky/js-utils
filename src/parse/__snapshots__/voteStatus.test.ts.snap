// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseVoteStatus should throw given an unknown value: "A" 1`] = `"Failed to parse vote status: A"`;

exports[`parseVoteStatus should throw given an unknown value: 100 1`] = `"Failed to parse vote status: 100"`;

exports[`parseVoteStatus should throw given an unknown value: null 1`] = `"Failed to parse vote status: null"`;
